### ISSUE TITLE
Custom scalar types

### ISSUE THREAD

**Issue by bboure** [aws/aws-appsync-community#26]
AppSync comes with the default GraphQL scalar types and with some handy [additional AWS types](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html).
It would be nice, however, to be able to add custom scalar types as well.

**Comment by mikeparisstuff** [comment]
Thank you for your comment. This feature request is in the backlog but I have noted this feedback for prioritization.

**Comment by vvenegasv** [comment]
This feature was developed?

**Comment by klyondx** [comment]
@mikeparisstuff kindly, is there some sort of ballpark eta on this feature?

**Comment by CorbinAmmanBF** [comment]
Is there an update on this feature request?

**Comment by Cliftonz** [comment]
Is their an update on this?

**Comment by th3m477** [comment]
+1 for this :)

**Comment by MontoyaAndres** [comment]
Any update?

**Comment by Danue1** [comment]
I'm here.
Any update?

**Comment by tglastra** [comment]
Is there any update on this issue? Is it still prioritised? Would very much like to have custom scalar types.

**Comment by ben-elsen** [comment]
I would be a fan of this feature as well!

**Comment by zerubeus** [comment]
3 years and still nothing done for this?

**Comment by joekendal** [comment]
Yes please

**Comment by fes300** [comment]
+1

**Comment by mirrorshades** [comment]
+1 ??

**Comment by samjett247** [comment]
+1

**Comment by joekendal** [comment]
Or at least rename AWSTimestamp, AWSDateTime etc. to Timestamp, DateTime. Revealing the cloud provider for customers is a potential security problem for that company.

**Comment by vknez** [comment]
This feature would make a difference for us. It would improve expressiveness of the schema, and on the other hand it would allow us to map the custom scalar type from the schema to the strongly typed module in the app. For example, we'd like to have a `CurrencyAmount` type that serializes to a `String`, but it is not the same to see:

```gql
... on Product {
  price: CurrencyAmount!
}
```

and 

```gql
... on Product {
  price: String!
}
```

Imagine now we set up our tooling to convert `CurrencyAmount` to whatever type would make our life easier in the app, certainly not a `string`.

**Comment by Cwellan** [comment]
> This feature would make a difference for us. It would improve expressiveness of the schema, and on the other hand it would allow us to map the custom scalar type from the schema to the strongly typed module in the app. For example, we'd like to have a `CurrencyAmount` type that serializes to a `String`, but it is not the same to see:
> 
> ```graphql
> ... on Product {
>   price: CurrencyAmount!
> }
> ```
> 
> and
> 
> ```graphql
> ... on Product {
>   price: String!
> }
> ```
> 
> Imagine now we set up our tooling to convert `CurrencyAmount` to whatever type would make our life easier in the app, certainly not a `string`.

I strongly agree with you. If we could match a custom scalar with a particular regex in order to efficiently check the validity of a given value, that would be great.

**Comment by irimiab** [comment]
We definitely need this feature; it actually is a blocker for us, when trying to adopt AppSync in our current architecture.

**Comment by ottosipe** [comment]
Custom scalars aside, I think at least adding support for `graphql-scalars` would probably make most of us happy.

https://www.graphql-scalars.dev/docs

**Comment by younesea** [comment]
+1

**Comment by Jrodna** [comment]
+1

**Comment by diegoNutmeg** [comment]
+1

**Comment by biggytech** [comment]
+1

**Comment by c-kle** [comment]
+1

**Comment by neaumusic** [comment]
+1

**Comment by leehambley** [comment]
+1, lack of custom scalars is a huge blocker for AppSync adoption for us, the whole purpose of GraphQL is expressive type system. We have a "no `String`" policy in our schema.

**Comment by veerendrapatel** [comment]
+1

**Comment by ivands** [comment]
+ 1

**Comment by zerubeus** [comment]
Happy new year guys, this is the most wanted feature.

**Comment by joekendal** [comment]
I think this is difficult for them to implement. So I don't see it coming anytime soon. 

Furthermore, I would make the case that Federation would be a much more impactful feature release. (Although it is technically supported, there is no support for a managed apollo router or federated schema in the console)

**Comment by nevingeorgesunny** [comment]
+ 1

**Comment by andresionek91** [comment]
Are there any workarounds for this issue?

**Comment by eduardo-caua** [comment]
+1

**Comment by NickSun** [comment]
+1

**Comment by ramonfallon** [comment]
+1

**Comment by ariksfaradi** [comment]
+1

**Comment by felipepxavier** [comment]
+1

**Comment by wireframe-sensei** [comment]
+1

**Comment by louislarry** [comment]
We want to have custom scalars so that:
- it matches our bounded context ubiquitous language
- to avoid primitive obsession
- we can map our typescript types into custom scalars

**Comment by Axum666** [comment]
+1

**Comment by austinamorusoyardstick** [comment]
I sort of made this work in a specific context for binary. Just built might be lots of bugs lol. https://www.npmjs.com/package/graphql-binary-transformer
https://github.com/austinamorusoyardstick/graphql-binary-transformer
Also only supports v1 transformer

**Comment by brendan-fox-shure** [comment]
This is also a blocker for us to adopt AppSync at the moment, as we need to implement a common Schema that required user defined types.  We would really love to avoid having to implement self hosted severless subscription if possible and leverage AppSync

**Comment by kniffo80** [comment]
+1

**Comment by ivorobioff** [comment]
> This is also a blocker for us to adopt AppSync at the moment, as we need to implement a common Schema that required user defined types. We would really love to avoid having to implement self hosted severless subscription if possible and leverage AppSync

avoid appsync, it's a trap

**Comment by steamstreetjon** [comment]
Seems like another one of those services (i.e. Cognito), that seems really useful, provides a unique value, but doesn't implement enough of the standards to allow us to use it as a true GraphQL service. It's 5 years later and nothing. A clear sign that there is minimal investment in the service.
